siriuspy/siriuspy/devices/bpm_eq.py

```diff
@@ -0,0 +1,541 @@
+        if ret > 0:
+            self._log(f'ERR: BPM {ret-1} did not start acquistion.')
+            return
+
+        self._log('Waiting BPMs to update')
+        ret = self.mturn_wait_update(timeout=self._acq_timeout)
+        if ret > 0:
+            self._log(f'ERR: BPM {ret-1} did not update in time.')
+            return
+        elif ret < 0:
+            self._log(f'ERR: Problem with acquisition. Error code {ret}')
+            return
+        self._log('BPMs updated.')
+
+        self._log('Acquiring data.')
+        fsamp = self.get_switching_frequency(1)
+        fswtc = self.get_sampling_frequency(1)
+        self.data['freq_switching'] = fsamp
+        self.data['freq_sampling'] = fswtc
+        if None in {fsamp, fswtc}:
+            self._log('ERR: Not all BPMs are configured equally.')
+            return
+        elif fsamp % fswtc:
+            self._log('ERR: Sampling freq is not multiple of switching freq.')
+            return
+
+        posx_gain, posy_gain = [], []
+        for bpm in self.bpms:
+            posx_gain.append(bpm.posx_gain)
+            posy_gain.append(bpm.posy_gain)
+        self.data['posx_gain'] = _np.array(posx_gain)
+        self.data['posy_gain'] = _np.array(posy_gain)
+        self.data['gains_acq'] = self.get_current_gains()
+
+        _time.sleep(0.1)
+        self.data['antennas'] = _np.array(
+            self.get_mturn_signals()).swapaxes(1, 0)
+
+    # --------- Methods for processing ------------
+
+    def process_data(self):
+        """Process data."""
+        if 'antennas' not in self.data:
+            self._log('ERR:There is no data to process. Acquire first.')
+            return
+        mean_antennas = self.calc_switching_levels(**self.data)
+        self.calc_gains(mean_antennas)
+        self.estimate_orbit_variation()
+
+    def calc_switching_levels(
+            self, antennas, fsamp=4, fswtc=2, acq_strategy=0,
+            acq_inverse_reduced_gain=0.95, **kwargs):
+        """Calculate average signal for each antenna in both switching states.
+
+        Args:
+            antennas (numpy.ndarray, (nrbpms, 4, N)): Antennas data.
+            fsamp (float, optional): Sampling frequency. Defaults to 4.
+            fswtc (float, optional): Switching frequency. Defaults to 2.
+            acq_strategy (int, optional): Whether we should assume states
+                order in data, starting with the direct state, or if we should
+                look for the higher levels (direct) and lower levels
+                (inverse). Defaults to 0.
+            acq_inverse_reduced_gain (float, optional): Level of the inverse
+                state, if acq_strategy is 1. Must be the same value used in
+                antennas gain during acquisition. Defaults to 0.95.
+
+        Returns:
+            mean (numpy.ndarray, nrbpms, 4, 2): The levels of each switching
+                state for each antenna.
+
+        """
+        _ = kwargs
```

Comment:
is kwargs necessary?

Reply:
I added it to make it easy to just pass `**self.data` as input. All other keys will be ignored in this way.